test/integration/src/test/java/io/pravega/test/integration/endtoendtest/AbstractEndToEndTest.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.integration.endtoendtest;
+
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.ReinitializationRequiredException;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.common.Exceptions;
+import io.pravega.segmentstore.contracts.StreamSegmentStore;
+import io.pravega.segmentstore.server.host.handler.PravegaConnectionListener;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.shared.segment.StreamSegmentNameUtils;
+import io.pravega.test.common.TestUtils;
+import io.pravega.test.common.TestingServerStarter;
+import io.pravega.test.common.ThreadPooledTestSuite;
+import io.pravega.test.integration.demo.ControllerWrapper;
+import lombok.Cleanup;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Before;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.junit.Assert.assertTrue;
+
+public class AbstractEndToEndTest extends ThreadPooledTestSuite {
```

Comment:
Can we add, perhaps separately, a system test for this stream cut feature?

Reply:
Sure, created an issue for it.